app/views/charges/new.html.erb

```diff
@@ -6,35 +6,108 @@
         </div>
+
+        <%= form.number_field :amount, step: 1, placeholder: '$0', class: 'input-field' %>
+
+        <%= recaptcha_tags %>
+        <button class="button primary">Make my donation</button>
       </div>
 
       <script>
-        const amountInput = document.getElementById('donation-amount');
+        // Create a Stripe client.
+        const stripe = Stripe('<%= ENV['STRIPE_PUBLISHABLE_KEY'] %>');
 
-        function setStripeCheckoutAmount(e) {
-          const amount = amountInput.value
-          const tag = document.getElementsByClassName('stripe-button')[0];
-          tag.dataset.amount = amount * 100; // amount in cents
-        }
+        // Create an instance of Elements.
+        const elements = stripe.elements();
 
-        amountInput.addEventListener('input', setStripeCheckoutAmount)
-      </script>
+        // Custom styling can be passed to options when creating an Element.
+        const style = {
+          base: {
+            color: '#32325d',
+            fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
+            fontSmoothing: 'antialiased',
+            fontSize: '16px',
+            '::placeholder': {
+              color: '#aab7c4'
+            }
+          },
+          invalid: {
+            color: '#fa755a',
+            iconColor: '#fa755a'
+          }
+        };
+
+        // Create an instance of the card Element.
+        const card = elements.create('card', {style: style});
+
+        // Add an instance of the card Element into the `card-element` <div>.
+        card.mount('#card-element');
 
-      <script src="https://checkout.stripe.com/checkout.js" class="stripe-button"
-              data-key="<%= ENV['STRIPE_PUBLISHABLE_KEY'] %>"
-              data-description="One time donation"
-              data-locale="auto"></script>
+        // Handle real-time validation errors from the card Element.
+        card.addEventListener('change', function(event) {
+          const displayError = document.getElementById('card-errors');
+          if (event.error) {
+            displayError.textContent = event.error.message;
+          } else {
+            displayError.textContent = '';
+          }
+        });
+
+        // Handle form submission.
+        const form = document.getElementById('payment-form');
+        form.addEventListener('submit', function(event) {
+          event.preventDefault();
+
+          stripe.createToken(card).then(function(result) {
+            if (result.error) {
+              // Inform the user if there was an error.
+              const errorElement = document.getElementById('card-errors');
+              errorElement.textContent = result.error.message;
+            } else {
+              // Send the token to your server.
+              stripeTokenHandler(result.token);
+            }
+          });
+        });
+
+        // Submit the form with the token ID.
+        function stripeTokenHandler(token) {
```

Comment:
@castrolem looks good for now, later we should move this functionality to a React component